app/src/main/java/com/example/android/weatherapp/fragment/BaseFragment.kt

```diff
@@ -1,7 +1,57 @@
 package com.example.android.weatherapp.fragment
 
+import android.view.View
+import android.widget.ProgressBar
+import android.widget.TextView
 import androidx.fragment.app.Fragment
+import com.example.android.weatherapp.utilities.MAX_PROGRESS
+import com.example.android.weatherapp.utilities.MIN_PROGRESS
 
-open class BaseFragment() : Fragment() {
+open class BaseFragment : Fragment() {
 
+    fun checkError(errorOccurred: Boolean, progressBar: ProgressBar, errorView: TextView,
+                 errorString: String) {
+        if (errorOccurred) {
+            hideProgressBar(progressBar)
+            showError(errorView, errorString)
+        } else {
+            showProgressBar(progressBar)
+            hideError(errorView)
+        }
+    }
+
+    fun checkLoading(isLoaded: Boolean, progressBar: ProgressBar) {
```

Comment:
checkLoading is not an appropriate name. Naming things is important as it makes the code more readable.